crates/service/src/index/mod.rs

```diff
@@ -31,8 +31,12 @@ use uuid::Uuid;
 use validator::Validate;
 
 #[derive(Debug, Error)]
-#[error("The index view is outdated.")]
-pub struct OutdatedError;
+pub enum InsertError {
```

Comment:
`Outdated` is an error for programming. `ServiceError` is an error for users and a fault for programming. Do not put them together.